### ISSUE TITLE
Major Exam Syllabus

### ISSUE THREAD

**Issue by pushpendradahiya** [vineetm/ell-881-2018-deep-learning#53]
@vineetm Sir,
What will be the syllabus for the major exam?
@raghavsi sir has given some pointers for CNN part.

**Comment by vineetm** [comment]
@pushpendradahiya Thank you for asking this. Syllabus for major exam would be exactly what was covered in the class. You are not expected to memorize or reproduce Tensorflow code/API. I will reiterate it here:

1. Linear Algebra: as covered in the class
2. Deep feedforward networks: understanding what is MLP, why non-linear layers help and  draw computational graphs as discussed in the class.
3. NLP Basics: Understanding the basic NLP pipeline. Why we need to convert a word to an integer. You got lot of practise for this as part of your project
4. Word2Vec: Understand the model of Word2Vec and the problem it addresses. 
5. RNN: What does a RNN model? The challenges involved in modeling such network. Understanding the need for gradient clipping. You are also expected to understand what is meant by perplexity and basics of Language Model.

Other than what I explicitly mentioned above, anything discussed in the lectures is fair game for the major exam.

**Comment by pushpendradahiya** [comment]
Also, 
Can the exam be open book, i.e. open laptop(without internet).
This technique is followed by many courses in IITD so that students don't have to mug up everything and importance is to understanding underlying concepts.

**Comment by vineetm** [comment]
No, the exam is not open book/laptop. 

This course does not expect/require you to mug up anything.

**Comment by pushpendradahiya** [comment]
Sure sir, thank you.
